### ISSUE TITLE
[Feature] Allow setting postgres port for integration tests

### ISSUE THREAD

**Issue by flexponsive** [dbt-labs/dbt-core#6795]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Currently, it is not possible to configure the port used by postgres in either `make setup-db` or `make integration`. It would be very useful if one could override the default port e.g. by passing a make parameter or using an environment variable, so one can run dbt integration tests without interfering with a postgres instance one may already have installed locally.

After a quick check, this looks like a relatively small feature to implement. There is already support for a `POSTGRES_TEST_PORT` environment variable in almost all of the test suite, so mainly small changes to the Makefile and docker-compose.yml would be required.

### Describe alternatives you've considered

*Leaving as is.* This works, just causes a bit of extra friction: stop local postgres before starting work on dbt-core, then restart the local postgres after done.

### Who will this benefit?

People wanting to contribute to `dbt-core` who already have postgres installed locally

### Are you interested in contributing this feature?

Yes

**Comment by dbeatty10** [comment]
Hey @flexponsive! 👋 

We'd welcome being able to customize the Postgres port so all the `make` commands can pick it up!

Please reach out to me when you have a PR ready or if you need any assistance along the way!

**Comment by flexponsive** [comment]
Hi @dbeatty10, thanks for the quick response! My pull request is ready for review

**Comment by jtcohen6** [comment]
This is already supported via env vars for the default Postgres connection, used for the vast majority of functional tests:

https://github.com/dbt-labs/dbt-core/blob/b2ea2b8b256e5db1da0b712dfedd7973e1e50a37/core/dbt/tests/fixtures/project.py#L114-L124

However, there are a number of functional tests where we override the Postgres connection details, e.g. to connect to a mixed-case database name (`dbtMixedCase`).

https://github.com/dbt-labs/dbt-core/blob/b2ea2b8b256e5db1da0b712dfedd7973e1e50a37/tests/functional/basic/test_copy_uppercase.py#L23-L42

I think the simplest fix is to use that same logic (`os.getenv("POSTGRES_TEST_PORT", 5432)`) in all tests where we're overriding the `dbt_profile_target` fixture.

@flexponsive I see your PR does something _similar_ to this, though with a bit more going on. Is there a reason you opted for  the approach you did?

**Comment by flexponsive** [comment]
@jtcohen6 i agree, when overriding the `dbt_profile_target` fixture it is better to follow the same approach as in `project.py` - and I will update the PR accordingly. For consistency, should I proceed analogously for `user`,`pass` and - where appropriate, e.g. `test_simple_dependency.py` - for `dbname` in the these cases?

This leaves open the question about assertions - should we assert against the `dbt_profile_target` like currently in the PR or also include `os.getenv("POSTGRES_TEST_PORT", 5432)` there? Example:

https://github.com/dbt-labs/dbt-core/blob/cb0fe563072963b2e775f06ac9627bd7683260b6/tests/functional/context_methods/test_env_vars.py#L154
